File: src/leetcode/weekComp/250601/Q2.cpp
#include <algorithm>
#include <iostream>
#include <limits.h>
#include <math.h>
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
  vector<vector<int>> minAbsDiff(vector<vector<int>> &grid, int k) {
    int m1 = grid.size(), n1 = grid[0].size();
    int m2 = m1 - k + 1, n2 = n1 - k + 1;
    vector<vector<int>> result(m2, vector<int>(n2, 0));
    if (k == 1) {
      return result;
    }
    for(int i = 0; i < m2; i++){
        for(int j = 0; j < n2; j++){
            
        }
    }
  }
};

int main() {
  // 示例二叉树
  Solution solution;
  vector<vector<int>> nums = {{1, 8}, {3, -2}};
  vector<vector<int>> result = solution.minAbsDiff(nums, 2);
  //    std::cout << "result: " << result << std::endl;
}

File: leetcode/weekComp/250601/Q2.py
from typing import List
from queue import Queue


class Solution:
    def minAbsDiff(self, grid: List[List[int]], k: int) -> List[List[int]]:
        m1 = len(grid)
        n1 = len(grid[0])
        m2 = m1 - k + 1
        n2 = n1 - k + 1
        result = [[0 for i in range(n2)] for j in range(m2)]
        if k == 1:
            return result

        def findClosetValue(mat):
            # 将子矩阵所有元素拉平成一维数组
            arr = []
            for row in mat:
                arr.extend(row)
            arr = set(arr)
            if len(arr) == 1:
                return 0
            arr = list(arr)
            arr.sort()
            min_diff = float("inf")
            # 找最小绝对差
            for i in range(1, len(arr)):
                min_diff = min(min_diff, abs(arr[i] - arr[i - 1]))
            return min_diff

        for i in range(m2):
            for j in range(n2):
                mat = [row[j : j + k] for row in grid[i : i + k]]
                result[i][j] = findClosetValue(mat)
        return result


if __name__ == "__main__":
    solution = Solution()
    grid = [[1, -2, 3], [2, 3, 5], [4, 5, 7]]
    result = solution.minAbsDiff(grid, 2)
    print(f"result = {result}")
